### ISSUE TITLE
New feature: list of most recent templates (write / fwd & reply) for template menus

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#263]
I notice that when I use the message header button dropdown menu to follow up with templates of a particular category, the complexity of selecting a single template is quite taxing, even when using keyboard shortcuts. See also issue #262 which will add accelerators for the actions (forward, reply, reply all, reply list) Therefore it would be great to add a list of the last N actions to the top level of the menu. So these would include the kind of follow up (forward / reply etc) and the template. They could be accessible through a simple number shortcut and would significantly simplify and speed up operations when writing many emails in a day based on a small number of templates. 

Example: in an sales email campaign reminding users about license renewals, I have to select a template from a number of choices depending on license type and location. FOr each of these selections I need to go through 4 separate selection events (pressing the ST button, selecting Forward, Selecting Category "SmartTemplates", then selecting the template after figuring out which of these cases applies:

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/dbb9000f-ca3b-4144-8095-979056c836ef)

Really we should be able to skip the first 3 steps and go straight to the selection of the type, which would be easily available when showing the last 10 used cases at the top level of the menu (ideally underneath the bottom menu item "settings")

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/94a7bf96-12ab-47c7-a86e-17b1f9633fb2)

**Comment by RealRaven2000** [comment]
Here is a first test version:

[smartTemplate-fx-4.3pre25.zip](https://github.com/RealRaven2000/SmartTemplates/files/12841438/smartTemplate-fx-4.3pre25.zip)

I have decided to only add "write template" items in the unified toolbar menu, and only "reply" and "forward" items in the toolbar header, to reflect the contextually most used items. These items are currently only built during the Thunderbird session (so they get lost when Thunderbird is restarted)

Example for the header area (Forward and Reply entries only)

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/2f962167-6d66-4597-a74b-7fba55a0b8b4)


---

To test the version above, download the zip file, drag the file into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by RealRaven2000** [comment]
New version to test - I am still intending to add file support so the recent menu items survive the session:

[smartTemplate-fx-4.3pre59.zip](https://github.com/RealRaven2000/SmartTemplates/files/12874220/smartTemplate-fx-4.3pre59.zip)

what's new:
- distinguishes between reply, reply all and reply list
- the main menu only gets write entries and the header menu only reply & forward entries
- now also supports adding items that are loaded ad-hoc (loading directly from disk without being in an existing menu entry
- added some icon styling for the template items, when hovering.

**Comment by RealRaven2000** [comment]
After prerelease 56, the MRU menu items didn't fire the event anymore, I found out this was caused by the following lines which add a "command" attribute. This obviously stopped the "oncommand" event firing:

```js
    if (theTemplate.command) {
      menuitem.setAttribute("command", theTemplate.command)
    }
```

I was using the attribute for finding out the type of reply (reply, replyList or replyAll) and now fixed the problem by prepending an underscore to the attribute, so that the command event handler can fire again.

[smartTemplate-fx-4.3pre70.zip](https://github.com/RealRaven2000/SmartTemplates/files/12904258/smartTemplate-fx-4.3pre70.zip)

----

To test the version above, download the zip file, drag the file into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by RealRaven2000** [comment]
Latest version below.

- fixed adding "open from file" function
- added saving / loading the mru list automatically. (the list is only loaded with a valid standard or pro license)
- maximum items / menu: 10 for Pro license (but can be edited), 5 for Standard license, 3 for no license.


[smartTemplate-fx-4.3pre78.zip](https://github.com/RealRaven2000/SmartTemplates/files/12907815/smartTemplate-fx-4.3pre78.zip)

-----

To test the version above, download the zip file, drag the file into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by RealRaven2000** [comment]
Implemented in 4.3.1 - published 16/10/2023
